onnxruntime/core/providers/coreml/builders/impl/builder_utils.cc

```diff
@@ -261,9 +261,8 @@ MILSpec::DataType OnnxDataTypeToMILSpec(int onnx_type) {
     case ONNX_NAMESPACE::TensorProto_DataType_INT16:
       return MILSpec::DataType::INT16;
     case ONNX_NAMESPACE::TensorProto_DataType_INT32:
-      return MILSpec::DataType::INT32;
     case ONNX_NAMESPACE::TensorProto_DataType_INT64:
-      return MILSpec::DataType::INT64;
+      return MILSpec::DataType::INT32;
```

Comment:
I think it may be more understandable to keep this helper function as a direct, one to one mapping.

a side question is, why do they even have the `MILSpec::DataType::INT64` data type?

Reply:
> a side question is, why do they even have the `MILSpec::DataType::INT64` data type?

easy to have a 'we might want this in the future' value. but AFAIK there is no usage of it in an actual MLModel.

but having a 1:1 mapping would be nicer as we could have an explicit comment on the case for INT64 saying that always gets converted to int32 in the CoreML world so the next developer to read the code understands what's going on.

Reply:
ok. when I see this function `OnnxDataTypeToMILSpec`, my initial assumption is that it just maps one enum type to another. if the destination enum also has "int64", it is unexpected (to me) that the source enum "int64" maps to something else.

perhaps a good place for describing this special behavior would be the comment that documents the function.
https://github.com/microsoft/onnxruntime/blob/f76e4de3b45b13204e69fc851201848803dc1b3a/onnxruntime/core/providers/coreml/builders/impl/builder_utils.h#L99-L101

Reply:
Ah, that's a fair point -- I added comment to both the header file and the switch case